File: scripts/update_icons.js
const fs = require('fs');
const path = 'c:/laragon/www/aiagent/src/components/icons/index.tsx';
let content = fs.readFileSync(path, 'utf8');

// Replace all instances of `export function XXIcon() {` 
// with `export function XXIcon({ size, className }: { size?: number, className?: string } = {}) {`
content = content.replace(/export function (\w+Icon)\(\) {/g, (match, name) => {
  if (name === 'WhatsAppIcon' || name === 'TelegramIcon') return match;
  return `export function ${name}({ size, className }: { size?: number, className?: string } = {}) {`;
});

// For each SVG tag, find width="XX" height="XX" and replace with width={size || XX} height={size || XX}
content = content.replace(/<svg\s+width="(\d+)"\s+height="(\d+)"\s+([^>]*)>/g, (match, w, h, rest) => {
  // also insert className={className} 
  if (rest.includes('className=')) {
    return `<svg width={size || ${w}} height={size || ${h}} ${rest}>`;
  }
  return `<svg width={size || ${w}} height={size || ${h}} className={className} ${rest}>`;
});

fs.writeFileSync(path, content, 'utf8');
console.log('Done!');

File: scripts/update_icons.py
import re
content = open('c:/laragon/www/aiagent/src/components/icons/index.tsx').read()

def replace_icon(match):
    name = match.group(2)
    if 'WhatsAppIcon' in name or 'TelegramIcon' in name:
        return match.group(0) # already has size
    
    # Extract the original width and height to use as default if we want, but letting default be the original is good.
    w_match = re.search(r'width="(\d+)"', match.group(0))
    orig_w = w_match.group(1) if w_match else "18"
    
    # replace first line
    res = re.sub(r'export function ' + name + r'\(\)', f'export function {name}({{ size = {orig_w}, className }}: {{ size?: number, className?: string }} = {{}})', match.group(0))
    # replace width and height
    res = re.sub(r'width="\d+"', 'width={size}', res, count=1)
    res = re.sub(r'height="\d+"', 'height={size}', res, count=1)
    
    # inject className
    # find the end of the opening svg tag and insert className={className} before it
    res = res.replace('viewBox="0 0 24 24"', 'className={className} viewBox="0 0 24 24"')
    return res

new_content = re.sub(r'(export function (\w+Icon)\(\) \{([\s\S]*?)<\/svg>\n\})', replace_icon, content)

with open('c:/laragon/www/aiagent/src/components/icons/index.tsx', 'w') as f:
    f.write(new_content)
